prov/shm/src/smr_ep.c

```diff
@@ -1259,6 +1259,12 @@ static void smr_init_ipc_socket(struct smr_ep *ep)
 static int smr_discard(struct fi_peer_rx_entry *rx_entry)
 {
 	struct smr_cmd_ctx *cmd_ctx = rx_entry->peer_context;
+	struct smr_region *peer_smr;
+	struct smr_resp *resp;
+
+	peer_smr = smr_peer_region(cmd_ctx->ep->region, cmd_ctx->cmd.msg.hdr.id);
+	resp = smr_get_ptr(peer_smr, cmd_ctx->cmd.msg.hdr.src_data);
+	resp->status = SMR_STATUS_SUCCESS;
```

Comment:
Good catch on this one! The response is not always valid though. In the inline and path it's not pointing to anything and the inject path it's pointing to an inject buffer which could result in a corruption somewhere. This needs to be
```
if (cmd_ctx->cmd_msg.hdr.src_data >= smr_src_iov) {
    //update resp
}
```
or something along those lines